nservicebus/upgrades/nhibernate-timeouts-index.md

```diff
@@ -0,0 +1,110 @@
+
+
+## Compatibility
+
+This issues has been resolved in following patch versions as defined in our  [support policy](support-policy.md):
+
+- 7.1.4 in use by NServiceBus 6.x
+- 7.0.6 (support 7.0.x) in use by NServiceBus 5.x
+- 6.2.8 (support 6.2.x) in use by NServiceBus 5.x
+
+
+
+If you are using any of the supported minor versions (7.1.x, 7.0.x, or 6.2.8) then you should at least update to the latest patch release. If you are using an older version then you should update to a newer minor (in case of 6.1.x or 6.0.x) or major version (any version prior to 6.x).
+
+
+## Upgrade steps
+
+Steps:
+
+ * Update to latest patch release
+ * Deploy the new version
+ * Check if you get a warning related to this schema issue
+   * Or manually inspect the schema in your database
+ * Follow procedure on how to resolve schema issues for your database engine (Microsoft SQL Server or Oracle)
+   * If you are using any other database engine then you have to apply these changes in the database dialect for your engine manually
+
+
+## Check at startup
+
+If you have endpoints with an incorrect table schema then this is detected in all fixed supported versions (at March 1th, 2017) for 6.2.x, 7.0.x and 7.1.x. The detection routine is run when the endpoint instance is created and started. If you are affected you will get the following log event with log level warning:
+
+> Could not find TimeoutEntity_EndpointIdx index. This may cause significant performance degradation of message deferral. Consult NServiceBus NHibernate persistence documentation for details on how to create this index.
+
+If this log event is written to your log file then read the following guidance on how to apply corrections.
+
+
+## Potential issues
+
+Any of the following issues can be present:
+
+- table `TimeoutEntity` has a clustered primary key
+- index `TimeoutEntity_EndpointIdx` is non-clustered
+- index `TimeoutEntity_EndpointIdx` has an incorrect column order (should be Endpoint, Time)
+
+
+How to correct these depend on the database engine that you are using.
+
+
+## Resolving schema issues with Microsoft SQL Server
+
+This assumes that both the index column order and clustered index are incorrect. To resolve this we will drop all existing indexes and recreate them.
+
+WARN: This procedure requires downtime. Make sure that all affected endpoint instances are not running or that the database is running in admin mode. This is needed because we are dropping index that guarantee consistency.
+
+NOTE: Make sure that the correct database is selected. If you are using a custom schema name then update the dbo schema with your custom schema identifier.
+
+NOTE: Please first run this script on your testing stage first in order to verify the script works as expected to reduce downtime.
+
+```sql
+BEGIN TRAN
+
+DROP INDEX [TimeoutEntity_SagaIdIdx] ON [dbo].[TimeoutEntity];
```

Comment:
I'm not familiar with what controls the schema selection with NHibernate. Is it an endpoint level config? The parameterization of schema makes the script harder to grok, so it might make more sense to have an INFO to adjust dbo to whatever schema you happen to be using.